### DISCUSSION TITLE
video 34 - seeking help: recipe not found

### DISCUSSION THREAD

**Discussion by jumarko** [jacekschae/learn-pedestal-course-files#25] [Q&A] [answered]
I cannot make this work in my code and I would appreciate some help.
I tried to debug it but failed to see why it's not working.
The code is here: https://github.com/jumarko/learn-pedestal/tree/develop/

The [retrieve recipe test](https://github.com/jumarko/learn-pedestal/blob/develop/src/test/cheffy/recipes_test.clj#L49-L55)
is failing because the the app returns nothing.
The trouble is that the db query returns empty result and I do not know why.
The query interceptor is implemented here: https://github.com/jumarko/learn-pedestal/blob/develop/src/main/cheffy/interceptors.clj#L35-L43

I added some logs: 
```
[nREPL-session-28d22a19-fe84-4895-8f90-489ae54f6ec0] DEBUG cheffy.interceptors - {:q-data {:query [:find (pull ?e pattern) :in $ ?recipe-id pattern :where [?e :recipe/recipe-id ?recipe-id]], :args [#uuid "baf9aea7-db4a-4bac-a503-f1d2c047ede0" [:recipe/recipe-id :recipe/prep-time :recipe/display-name :recipe/image-url :recipe/public? :account/_favorite-recipes #:recipe{:owner [:account/account-id :account/display-name]} #:recipe{:steps [:step/step-id :step/description :step/sort-order]} #:recipe{:ingredients [:ingredient/ingredient-id :ingredient/display-name :ingredient/amount :ingredient/measure :ingredient/sort-order]}]]}, :database #datomic.core.db.Db{:id "c4b88287-b125-430b-9b5a-c876a3428664", :basisT 26, :indexBasisT 0, :index-root-id nil, :asOfT nil, :sinceT nil, :raw nil}, :line 37}
[nREPL-session-28d22a19-fe84-4895-8f90-489ae54f6ec0] TRACE cheffy.interceptors - {:q-data-with-database {:query [:find (pull ?e pattern) :in $ ?recipe-id pattern :where [?e :recipe/recipe-id ?recipe-id]], :args [#datomic.core.db.Db{:id "c4b88287-b125-430b-9b5a-c876a3428664", :basisT 26, :indexBasisT 0, :index-root-id nil, :asOfT nil, :sinceT nil, :raw nil} #uuid "baf9aea7-db4a-4bac-a503-f1d2c047ede0" [:recipe/recipe-id :recipe/prep-time :recipe/display-name :recipe/image-url :recipe/public? :account/_favorite-recipes #:recipe{:owner [:account/account-id :account/display-name]} #:recipe{:steps [:step/step-id :step/description :step/sort-order]} #:recipe{:ingredients [:ingredient/ingredient-id :ingredient/display-name :ingredient/amount :ingredient/measure :ingredient/sort-order]}]]}, :line 39}
[nREPL-session-28d22a19-fe84-4895-8f90-489ae54f6ec0] TRACE cheffy.interceptors - {:q-result [], :line 42}
```

I also verified that `:basisT` is as expected - e.g., it was 19 before `transcact!` was executed and 20 afterward. 
It was also 20 when it entered the `query!` function.

**Comment by jumarko** [comment, marked as answer]
After a long struggle and migrating to the peer library (datomic) I managed to fix the problem: https://github.com/jumarko/learn-pedestal/pull/19

There were a couple of problems: 
- tx-data was not wrapped in a map (key `:tx-data`) but passed as a top-level arg
   - notice that this is actually the difference between the peer library (`datomic.api`) and the client library (`datomic.client.api`)
- returning a bare ring response from the route handler instead of the whole context.

**Reply by jacekschae** [reply]
Glad to hear that you figured it out! 👏
